examples/webgl_loader_gltf.html

```diff
@@ -1,642 +1,170 @@
-				scene.add( defaultCamera );
-				camera = defaultCamera;
+			function init() {
 
-				var sceneInfo = sceneList[index];
+				container = document.createElement( 'div' );
+				document.body.appendChild( container );
 
-				var spot1 = null;
+				camera = new THREE.PerspectiveCamera( 45, window.innerWidth / window.innerHeight, 0.25, 20 );
+				camera.position.set( -1.8, 0.9, 2.7 );
 
-				if (sceneInfo.addLights) {
+				controls = new THREE.OrbitControls( camera );
+				controls.target.set( 0, -0.2, -0.2 );
+				controls.update();
 
-					var ambient = new THREE.AmbientLight( 0x222222 );
-					scene.add( ambient );
+				scene = new THREE.Scene();
+				scene.background = new THREE.Color( 0x443333 );
+				scene.fog = new THREE.Fog( 0x443333, 2, 10 );
+
+				light = new THREE.HemisphereLight( 0xffffff, 0x444444 );
+				light.position.set( 0, 2, 0 );
+				scene.add( light );
+
+				light = new THREE.DirectionalLight( 0xffffff );
+				light.position.set( 0, 2, 1 );
+				light.castShadow = true;
+				light.shadow.camera.top = 1.8;
+				light.shadow.camera.bottom = -1.8;
+				light.shadow.camera.left = -1.2;
+				light.shadow.camera.right = 1.2;
+				scene.add( light );
+
+				// ground
+				var plane = new THREE.Mesh(
+					new THREE.PlaneBufferGeometry( 20, 20 ),
+					new THREE.MeshPhongMaterial( { color: 0x999999, specular: 0x101010 } )
+				);
+				plane.rotation.x = - Math.PI / 2;
+				plane.position.y = -1;
+				plane.receiveShadow = true;
+				scene.add(plane);
+
+				var grid = new THREE.GridHelper( 20, 20, 0x000000, 0x000000 );
+				grid.position.y = -0.99;
+				grid.material.opacity = 0.2;
+				grid.material.transparent = true;
+				scene.add( grid );
+
+				// envmap
+				var path = 'textures/cube/Park2/';
+				var format = '.jpg';
+				var envMap = new THREE.CubeTextureLoader().load( [
+					path + 'posx' + format, path + 'negx' + format,
+					path + 'posy' + format, path + 'negy' + format,
+					path + 'posz' + format, path + 'negz' + format
+				] );
 
-					var directionalLight = new THREE.DirectionalLight( 0xdddddd );
-					directionalLight.position.set( 0, 0, 1 ).normalize();
-					scene.add( directionalLight );
+				// model
+				var loader = new THREE.GLTFLoader();
+				loader.load( 'models/gltf/DamagedHelmet/glTF/DamagedHelmet.gltf', function ( gltf ) {
 
-					spot1   = new THREE.SpotLight( 0xffffff, 1 );
-					spot1.position.set( 10, 20, 10 );
-					spot1.angle = 0.25;
-					spot1.distance = 1024;
-					spot1.penumbra = 0.75;
+					gltf.scene.traverse( function ( child ) {
 
-					if ( sceneInfo.shadows ) {
+						if ( child.isMesh ) {
 
-						spot1.castShadow = true;
-						spot1.shadow.bias = 0.0001;
-						spot1.shadow.mapSize.width = 2048;
-						spot1.shadow.mapSize.height = 2048;
+							child.material.envMap = envMap;
+							child.material.needsUpdate = true;
+							child.castShadow = true;
 
-					}
+						}
 
-					scene.add( spot1 );
+					} );
 
-				}
+					scene.add.apply( scene, gltf.scene.children );
```

Comment:
Not strong preference but I prefer

    scene.add( gltf.scene );

because I'm really not sure how well `function.apply` style is known to JS (especially new) users.

If we don't wanna push `gltf.scene` to `scene`, the code would be like this?

or 

    while( gltf.scene.children.length > 0 ) {

        scene.add( gltf.scene.children[ 0 ] );

    }

(a bit complicated?)

Reply:
Sure ✅